### PR TITLE
Add remove data access entries from Scope

### PR DESCRIPTION
**Pull request by channa-figure**
## Description

Adding rpc and cli command to remove and add data access addresses to scopes.

closes: #220 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/metadata/client/cli/tx.go [hunk 1]
```diff
@@ -155,6 +156,58 @@ func RemoveScopeCmd() *cobra.Command {
 	return cmd
 }
 
+func AddRemoveScopeDataAccessCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "scope-data-access [add|remove] [scope-id] [data-access]",
```
Comment (dwedul-figure) [inline x/metadata/client/cli/tx.go:161]: For the `Use` lines, the documentation says:

> Use is the one-line usage message. Recommended syntax is as follow:
  [ ] identifies an optional argument. Arguments that are not enclosed in brackets are required.
  ... indicates that you can specify multiple values for the previous argument.
  |   indicates mutually exclusive information. You can use the argument to the left of the separator or the
      argument to the right of the separator. You cannot use both arguments in a single use of the command.
  { } delimits a set of mutually exclusive arguments when one of the arguments is required. If the arguments are
      optional, they are enclosed in brackets ([ ]).
Example: add [-F file | -D dir]... [-f format] profile

```suggestion
		Use:   "scope-data-access {add|remove} scope-id data-access",
```

I updated the other `Use` lines in this file in my other PR.

File: proto/provenance/metadata/v1/tx.proto [hunk 2]
```diff
@@ -122,6 +127,52 @@ message MsgDeleteScopeRequest {
...
+  option (gogoproto.goproto_getters)  = false;
+
+  // scope MetadataAddress for removing data access
+  bytes scope_id = 1 [
+    (gogoproto.nullable)   = false,
+    (gogoproto.customtype) = "MetadataAddress",
+    (gogoproto.moretags)   = "yaml:\"scope_id\""
+  ];
+
+  // MetadataAddresses to be removed from scope
```
Comment (dwedul-figure) [inline proto/provenance/metadata/v1/tx.proto:167]: Nit: These are bech32 address strings, not necessarily `MetadataAddress`es.

File: x/metadata/client/cli/tx.go (hunk 1 above)
Comment (channa-figure) [reply x/metadata/client/cli/tx.go:161]: Got it, I was following what we had before.  We have this all over the place with the [] on required args.

File: proto/provenance/metadata/v1/tx.proto (hunk 2 above)
Comment (channa-figure) [reply proto/provenance/metadata/v1/tx.proto:167]: 🤦 Yep, you are right.  Thanks!

File: proto/provenance/metadata/v1/tx.proto [hunk 3]
```diff
@@ -122,6 +127,52 @@ message MsgDeleteScopeRequest {
...
+  option (gogoproto.goproto_getters)  = false;
+
+  // scope MetadataAddress for removing data access
+  bytes scope_id = 1 [
+    (gogoproto.nullable)   = false,
+    (gogoproto.customtype) = "MetadataAddress",
+    (gogoproto.moretags)   = "yaml:\"scope_id\""
+  ];
+
+  // bech32 address to be removed from scope
```
Comment (iramiller) [inline proto/provenance/metadata/v1/tx.proto:167]: Not totally sold on the "bech32 addresses" title here... the next line just uses "address" to describe these values... if we were being more formal the term is "AccAddress"

**Review by iramiller — APPROVED** [review]

**Pull request merged**